server/datastore/mysql/scim.go

```diff
@@ -125,6 +126,91 @@ func (ds *Datastore) ScimUserByUserName(ctx context.Context, userName string) (*
 	return user, nil
 }
 
+// ScimUserByUserNameOrEmail finds a SCIM user by username. If it cannot find one, then it tries email, if set.
+// If multiple users are found with the same email, we log an error and return nil.
+// Emails and groups are NOT populated in this method.
+func (ds *Datastore) ScimUserByUserNameOrEmail(ctx context.Context, userName string, email *string) (*fleet.ScimUser, error) {
+	// First, try to find the user by userName
+	if userName != "" {
+		user, err := ds.ScimUserByUserName(ctx, userName)
+		switch {
+		case err == nil:
+			return user, nil
+		case !fleet.IsNotFound(err):
+			return nil, ctxerr.Wrap(ctx, err, "select scim user by userName")
+		}
+		if email == nil || *email == "" {
+			return nil, notFound("scim user")
+		}
+	}
+
+	// Try to find the user by email
+	const query = `
+		SELECT
+			scim_users.id, external_id, user_name, given_name, family_name, active, scim_users.updated_at
+		FROM scim_users
+		JOIN scim_user_emails ON scim_users.id = scim_user_emails.scim_user_id
+		WHERE scim_user_emails.email = ?
+	`
+
+	var users []fleet.ScimUser
+	err := sqlx.SelectContext(ctx, ds.reader(ctx), &users, query, *email)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
```

Comment:
`SelectContext` does not return `ErrNoRows`, only `GetContext` does, so you can safely remove this condition check.